terraform/modules/hub/matomo.tf

```diff
@@ -0,0 +1,536 @@
+        ]
+      }
+    ]
+  }
+EOF
+
+}
+
+resource "aws_iam_role" "matomo_cloudwatch_execution" {
+  name = "matomo_cloudwatch_execution"
+
+  assume_role_policy = <<DOC
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "events.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+DOC
+
+}
+
+resource "aws_iam_role_policy" "allow_run_ecs_tasks" {
+  name = "allow-run-ecs-tasks"
+  role = aws_iam_role.matomo_cloudwatch_execution.id
+
+  policy = <<DOC
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "iam:PassRole",
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "ecs:RunTask",
+            "Resource": "*"
+        }
+    ]
+}
+DOC
+
+}
+resource "aws_cloudwatch_event_target" "matomo_scheduled_archive_task" {
+  target_id = "run-matomo-archive-task"
+  arn       = aws_ecs_cluster.fargate-ecs-cluster.id
+  rule      = aws_cloudwatch_event_rule.every_hour.name
+  role_arn  = aws_iam_role.matomo_cloudwatch_execution.arn
+
+  ecs_target {
+    task_count          = 1
+    task_definition_arn = aws_ecs_task_definition.matomo_archiving_task_def.arn
+    launch_type         = "FARGATE"
+
+    network_configuration {
+      security_groups = [
+        aws_security_group.matomo_lb.id,
+        aws_security_group.matomo.id,
+      ]
+
+      subnets = aws_subnet.internal.*.id
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "matomo_execution_matomo_web_secrets" {
+  role       = aws_iam_role.matomo_execution.name
+  policy_arn = aws_iam_policy.matomo_web_secrets.arn
+}
+
+resource "aws_cloudwatch_event_rule" "every_hour" {
+  name                = "every-hour"
+  schedule_expression = "rate(1 hour)"
+}
+
+data "template_file" "matomo_config_part_one_file" {
+  template = file("${path.module}/files/matomo/matomo-config-file-part-one.ini.php")
+
+  vars = {
+    db_host      = aws_db_instance.matomo.address
+    db_password  = random_string.matomo_db_password.result
+    trusted_host = aws_route53_record.matomo.name
+    salt         = var.matomo_salt
+  }
+}
+
+resource "aws_ssm_parameter" "matomo_config_file_part_one" {
+  name        = "/matomo/config-file-part-one"
+  description = "Base64 encoded config file part one for Matomo"
+  type        = "SecureString"
+  key_id      = aws_kms_key.matomo.arn
+  value       = base64encode(data.template_file.matomo_config_part_one_file.rendered)
+}
+
+resource "aws_ssm_parameter" "matomo_config_file_part_two" {
+  name        = "/matomo/config-file-part-two"
+  description = "Base64 encoded config file part two for Matomo"
+  type        = "SecureString"
+  key_id      = aws_kms_key.matomo.arn
+  value       = base64encode(file("${path.module}/files/matomo/matomo-config-file-part-two.ini.php"))
+}
+
+resource "random_string" "matomo_db_password" {
+  length  = 41
```

Comment:
I have to ask... why 41 of all numbers? 😄

Reply:
according to git blame "Mysql doesn't like too long passwords" :D